### ISSUE TITLE
Galaxy startup fails (MacOS 10.13)

### ISSUE THREAD

**Issue by mcornudella** [galaxyproject/galaxy#5729]
Hi everyone,
I just downloaded galaxy following the instructions in https://galaxyproject.org/admin/get-galaxy/ and when I run run.sh I get the following error:

```
Activating virtualenv at .venv
Requirement already satisfied: pip>=8.1 in ./.venv/lib/python2.7/site-packages
Requirement already satisfied: numpy==1.9.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 4))
Requirement already satisfied: bx-python==0.7.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 5))
Requirement already satisfied: MarkupSafe==1.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 6))
Requirement already satisfied: PyYAML==3.12 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 7))
Requirement already satisfied: SQLAlchemy==1.0.15 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 8))
Requirement already satisfied: sqlalchemy-utils==0.32.19 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 9))
Requirement already satisfied: mercurial==3.7.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 10))
Requirement already satisfied: pycrypto==2.6.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 11))
Requirement already satisfied: uWSGI==2.0.15 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 12))
Requirement already satisfied: pysam==0.14 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 14))
Requirement already satisfied: bz2file==0.98 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 21))
Requirement already satisfied: ipaddress==1.0.18 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 22))
Requirement already satisfied: boltons==17.1.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 23))
Requirement already satisfied: Paste==2.0.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 24))
Requirement already satisfied: PasteDeploy==1.5.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 25))
Requirement already satisfied: docutils==0.14 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 26))
Requirement already satisfied: repoze.lru==0.6 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 27))
Requirement already satisfied: Routes==2.4.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 28))
Requirement already satisfied: WebOb==1.4.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 29))
Requirement already satisfied: WebHelpers==1.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 30))
Requirement already satisfied: Mako==1.0.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 31))
Requirement already satisfied: pytz==2017.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 32))
Requirement already satisfied: Babel==2.5.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 33))
Requirement already satisfied: Beaker==1.7.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 34))
Requirement already satisfied: dictobj==0.3.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 35))
Requirement already satisfied: nose==1.3.7 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 36))
Requirement already satisfied: Parsley==1.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 37))
Requirement already satisfied: six==1.11.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 38))
Requirement already satisfied: Whoosh==2.7.4 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 39))
Requirement already satisfied: galaxy_sequence_utils==1.0.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 40))
Requirement already satisfied: h5py==2.7.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 41))
Requirement already satisfied: packaging==16.8 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 42))
Requirement already satisfied: pykwalify==1.6.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 45))
Requirement already satisfied: python-dateutil==2.5.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 46))
Requirement already satisfied: docopt==0.6.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 47))
Requirement already satisfied: Cheetah==2.4.4 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 50))
Requirement already satisfied: Markdown==2.6.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 51))
Requirement already satisfied: requests==2.18.4 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 54))
Requirement already satisfied: certifi==2017.7.27.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 55))
Requirement already satisfied: urllib3==1.22 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 56))
Requirement already satisfied: chardet==3.0.4 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 57))
Requirement already satisfied: idna==2.6 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 58))
Requirement already satisfied: bioblend==0.7.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 61))
Requirement already satisfied: boto==2.38.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 62))
Requirement already satisfied: requests-toolbelt==0.8.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 63))
Requirement already satisfied: kombu==3.0.30 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 67))
Requirement already satisfied: amqp==1.4.8 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 68))
Requirement already satisfied: anyjson==0.3.3 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 69))
Requirement already satisfied: psutil==4.1.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 72))
Requirement already satisfied: pulsar-galaxy-lib==0.8.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 73))
Requirement already satisfied: sqlalchemy-migrate==0.11.0 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 76))
Requirement already satisfied: decorator==4.0.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 77))
Requirement already satisfied: Tempita==0.5.3dev in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 78))
Requirement already satisfied: sqlparse==0.1.16 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 79))
Requirement already satisfied: pbr==3.1.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 80))
Requirement already satisfied: svgwrite==1.1.6 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 83))
Requirement already satisfied: pyparsing==2.1.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 84))
Requirement already satisfied: Fabric==1.13.2 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 87))
Requirement already satisfied: paramiko==2.2.1 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 88))
Requirement already satisfied: ecdsa==0.13 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 89))
Requirement already satisfied: python-genomespaceclient==0.1.8 in ./.venv/lib/python2.7/site-packages (from -r requirements.txt (line 92))
Requirement already satisfied: PasteScript in ./.venv/lib/python2.7/site-packages (from pulsar-galaxy-lib==0.8.0->-r requirements.txt (line 73))
Requirement already satisfied: cryptography>=1.1 in ./.venv/lib/python2.7/site-packages (from paramiko==2.2.1->-r requirements.txt (line 88))
Requirement already satisfied: bcrypt>=3.1.3 in ./.venv/lib/python2.7/site-packages (from paramiko==2.2.1->-r requirements.txt (line 88))
Requirement already satisfied: pyasn1>=0.1.7 in ./.venv/lib/python2.7/site-packages (from paramiko==2.2.1->-r requirements.txt (line 88))
Requirement already satisfied: pynacl>=1.0.1 in ./.venv/lib/python2.7/site-packages (from paramiko==2.2.1->-r requirements.txt (line 88))
Requirement already satisfied: python-swiftclient in ./.venv/lib/python2.7/site-packages (from python-genomespaceclient==0.1.8->-r requirements.txt (line 92))
Requirement already satisfied: boto3 in ./.venv/lib/python2.7/site-packages (from python-genomespaceclient==0.1.8->-r requirements.txt (line 92))
Requirement already satisfied: cffi>=1.7; platform_python_implementation != "PyPy" in ./.venv/lib/python2.7/site-packages (from cryptography>=1.1->paramiko==2.2.1->-r requirements.txt (line 88))
Requirement already satisfied: enum34; python_version < "3" in ./.venv/lib/python2.7/site-packages (from cryptography>=1.1->paramiko==2.2.1->-r requirements.txt (line 88))
Requirement already satisfied: asn1crypto>=0.21.0 in ./.venv/lib/python2.7/site-packages (from cryptography>=1.1->paramiko==2.2.1->-r requirements.txt (line 88))
Requirement already satisfied: futures>=3.0; python_version == "2.7" or python_version == "2.6" in ./.venv/lib/python2.7/site-packages (from python-swiftclient->python-genomespaceclient==0.1.8->-r requirements.txt (line 92))
Requirement already satisfied: s3transfer<0.2.0,>=0.1.10 in ./.venv/lib/python2.7/site-packages (from boto3->python-genomespaceclient==0.1.8->-r requirements.txt (line 92))
Requirement already satisfied: botocore<1.10.0,>=1.9.11 in ./.venv/lib/python2.7/site-packages (from boto3->python-genomespaceclient==0.1.8->-r requirements.txt (line 92))
Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in ./.venv/lib/python2.7/site-packages (from boto3->python-genomespaceclient==0.1.8->-r requirements.txt (line 92))
Requirement already satisfied: pycparser in ./.venv/lib/python2.7/site-packages (from cffi>=1.7; platform_python_implementation != "PyPy"->cryptography>=1.1->paramiko==2.2.1->-r requirements.txt (line 88))
Activating virtualenv at .venv
galaxy.tools.deps DEBUG 2018-03-19 18:18:30,405 Unable to find config file './dependency_resolvers_conf.xml'
executing: .venv/bin/uwsgi --module 'galaxy.webapps.galaxy.buildapp:uwsgi_app()' --virtualenv /Users/miquelcornudella/Documents/galaxy/.venv --pythonpath lib --threads 4 --http localhost:8080 --static-map /static/style=/Users/miquelcornudella/Documents/galaxy/static/style/blue --static-map /static=/Users/miquelcornudella/Documents/galaxy/static --die-on-term --hook-master-start 'unix_signal:2 gracefully_kill_them_all' --hook-master-start 'unix_signal:15 gracefully_kill_them_all' --enable-threads --py-call-osafterfork 
[uwsgi-static] added mapping for /static/style => /Users/miquelcornudella/Documents/galaxy/static/style/blue
[uwsgi-static] added mapping for /static => /Users/miquelcornudella/Documents/galaxy/static
*** Starting uWSGI 2.0.15 (64bit) on [Mon Mar 19 18:18:30 2018] ***
compiled with version: 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.72) on 10 November 2017 07:29:21
os: Darwin-17.4.0 Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64
nodename: poblenou-128-243.eduroamcp.upf.edu
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /Users/miquelcornudella/Documents/galaxy
detected binary path: /Users/miquelcornudella/Documents/galaxy/.venv/bin/python
!!! no internal routing support, rebuild with pcre support !!!
*** WARNING: you are running uWSGI without its master process manager ***
your processes number limit is 1418
your memory page size is 4096 bytes
detected max file descriptor number: 256
building mime-types dictionary from file /etc/apache2/mime.types...1002 entry found
lock engine: OSX spinlocks
thunder lock: disabled (you can enable it with --thunder-lock)
uWSGI http bound on localhost:8080 fd 4
spawned uWSGI http 1 (pid: 80147)
uwsgi socket 0 bound to TCP address 127.0.0.1:56837 (port auto-assigned) fd 3
Python version: 2.7.12 (v2.7.12:d33e0cf91556, Jun 26 2016, 12:10:39)  [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

Any Idea how I could fix it in order to run galaxy?

Thanks in advance,
Miquel

**Comment by natefoo** [comment]
Is there an error message that appears after this?

**Comment by mcornudella** [comment]
Hi, thanks for trying to help me and sorry for not attaching the complete trace before (I thought I did)

After this message it enters into debug mode. It gets stuck at `galaxy.queue_worker INFO 2018-03-19 21:44:31,780 [p:80392,w:1,m:0] [MainThread] Binding and starting galaxy control worker for main.web.1`

I attach the complete trace.

Thanks again,
Miquel

*edit by @martenson*: moved gist to https://gist.github.com/martenson/9a6daa843e677eaa447aad0b7d757d59

**Comment by biomendi** [comment]
Same issue here on a Mac Pro running MacOs 10.12 (Sierra). 

Is anyone looking into this? 

Thanks a lot!

**Comment by mvdbeek** [comment]
@mcornudella so far it looks like it's working, did you try reaching galaxy (http://localhost:8080 is the default) ?

**Comment by mvdbeek** [comment]
@biomendi are you seeing any error ?

**Comment by nsoranzo** [comment]
I suppose that not showing these 2 lines in the log any more can easily cause confusion:
```
Starting server in PID 12786.
serving on http://127.0.0.1:8080
```
@natefoo Can we do that with uWSGI?

**Comment by natefoo** [comment]
We can generate log messages like that explicitly, I don't think uWSGI has any facility for doing it automatically like Paste. They're going to be in the galaxy log format (prefixed with `galaxy.foo.bar [...`) unless we use `print()` though (which is maybe not so bad in this specific case).

**Comment by mcornudella** [comment]
Thank you all!

After 'run.sh' I can reach galaxy at localhost:8080 . Sorry for the confusion!

**Comment by nsoranzo** [comment]
@mcornudella Thanks for the bug report any way, this raised awareness that the log file was missing this important notification, which will be fixed by pull request #5770.
